delegates/src/main/java/com/revolut/recyclerkit/delegates/DiffAdapter.kt

```diff
@@ -25,22 +28,113 @@ import androidx.recyclerview.widget.DiffUtil
  *
  */
 
-class DiffAdapter(
-    delegatesManager: DelegatesManager = DelegatesManager()
+open class DiffAdapter(
+    delegatesManager: DelegatesManager = DelegatesManager(),
+    async: Boolean = true,
+    autoScrollToTop: Boolean = false
 ) : AbsRecyclerDelegatesAdapter(delegatesManager) {
 
-    private val differ: AsyncListDiffer<ListItem> = AsyncListDiffer(
-        AdapterListUpdateCallback(this),
-        AsyncDifferConfig.Builder(ListDiffCallback<ListItem>()).build()
-    )
+    private interface DifferDelegate {
+        val items: List<ListItem>
+        fun setItems(items: List<ListItem>, recyclerView: RecyclerView)
+    }
+
+    private val differDelegate = if (async) AsyncDifferDelegate(this, autoScrollToTop) else SyncDifferDelegate(this, autoScrollToTop)
```

Comment:
Nice approach with delegating differ logic to the separate abstraction
Let's reuse `SyncDifferDelegate` in `RxDiffAdapter`. Otherwise we have duplicated logic in the parent and child classes

Reply:
You can also use naming like AsyncDifferStrategy and SyncDifferStrategy because those delegates represent strategy pattern